### ISSUE TITLE
Performance of AT is worse when using the Pytorch official ResNet-18 architecture.

### ISSUE THREAD

**Issue by futakw** [ndb796/Pytorch-Adversarial-Training-CIFAR#5]
This implementation works perfectly.

However, by only changing the architecture to the Pytorch official ResNet-18 (https://pytorch.org/vision/main/models/generated/torchvision.models.resnet18.html), the performance of PGD-AT is much lower.
Natural accuracy and robust accuracy are both around 10% lower.

Do you know what causes this difference?
Is there any reason that you use different architecture in this repository?

Thank you in advance.

===
Difference between models:
1.  
In this repo, the first layer is
`self.conv1 = nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False)`
whereas the torchvision model is
`self.conv1 = nn.Conv2d(3, self.inplanes, kernel_size=7, stride=2, padding=3, bias=False)`
2.
In this repo, after the blocks, avgpool is applied to vector.
`F.avg_pool2d(out, 4)`
whereas the torchvision model applies an adaptive avgpool.
`self.avgpool = nn.AdaptiveAvgPool2d((1, 1))`

**Comment by fabian-sp** [comment]
I am not an author of this repo, but the Pytorch models are for Imagenet - if I remember correctly. So the last layer size is larger than 10 (which is what is needed for CIFAR10) as imagenet has more classes.

**Comment by futakw** [comment]
Hi, I figured out why a while ago but forgot to resolve this issue.

As you said, Pytorch's official model is for ImageNet, which expects 224x224 resolution.
And when you want to try CIFAR-10, which is 32x32, the kernel size of the first convolution layer should be different.
And this is not only true for AT performance but also for natural accuracy: it would be much lower if the official Pytorch model is used for CIFAR-10.

Thank you for your reply!
